src/ol/layer/Layer.js

```diff
@@ -210,6 +210,9 @@ class Layer extends BaseLayer {
    * an array of features.
    */
   getFeatures(pixel) {
+    if (!this.renderer_) {
+      return new Promise((resolve) => resolve([]));
```

Comment:
Minor, but this could be `return Promise.resolve([])`.

Reply:
Thanks @tschaub, I did not know this. Should I create a PR?

Reply:
Not critical.